### PR TITLE
test: improve test

### PR DESCRIPTION
**Pull request by chokobole**
# Description

This PR enhances tests in several aspects:

- Minimizes build dependencies.
- Introduces `math::FiniteFieldTest` and `c::math::bn254::G1Test` to eliminate boilerplate code.
- And more.

### CODE REVIEW

File: tachyon/zk/lookup/compress_expression_unittest.cc [hunk 1]
```diff
@@ -9,12 +9,10 @@
...
 
 #include "tachyon/zk/expressions/expression_factory.h"
-#include "tachyon/zk/lookup/test/compress_expression_test_setting.h"
+#include "tachyon/zk/lookup/test/compress_expression_test.h"
 
 namespace tachyon::zk {
 
-class CompressExpressionTest : public CompressExpressionTestSetting {};
-
 TEST_F(CompressExpressionTest, CompressExpressions) {
```
Comment (ashjeong) [inline tachyon/zk/lookup/compress_expression_unittest.cc:16]: inherit `public CompressExpressionTest` and change the class name, or change `TEST_F` to `TEST`

Comment (chokobole) [reply tachyon/zk/lookup/compress_expression_unittest.cc]: I think we don't have to inherit the class in this case.

Comment (ashjeong) [reply tachyon/zk/lookup/compress_expression_unittest.cc:16]: In that case, change `TEST_F` to `TEST`?

**Review by TomTaehoonKim — APPROVED** [review]
Else, LGTM

File: tachyon/zk/lookup/compress_expression_unittest.cc (hunk 1 above)
Comment (chokobole) [reply tachyon/zk/lookup/compress_expression_unittest.cc:16]: This still needs a fixture, otherwise it won't get compield.

**Review by ashjeong — APPROVED** [review]
LGTM

File: tachyon/c/math/polynomials/univariate/bn254_univariate_evaluation_domain_unittest.cc [hunk 2]
```diff
@@ -70,7 +70,7 @@ TEST_F(UnivariateEvaluationDomainTest, FFT) {
           domain_,
           reinterpret_cast<const tachyon_bn254_univariate_dense_polynomial*>(
               &poly));
-  EXPECT_EQ(Domain::Create(kDegree)->FFT(poly),
+  EXPECT_EQ(Domain::Create(kDegree + 1)->FFT(poly),
```
[the remark this answers is not in the document]
Comment (chokobole) [reply tachyon/c/math/polynomials/univariate/bn254_univariate_evaluation_domain_unittest.cc:72]: It gets ceiled.

```suggestion
  static std::unique_ptr<Radix2EvaluationDomain> Create(size_t num_coeffs) {
    return absl::WrapUnique(new Radix2EvaluationDomain(
        absl::bit_ceil(num_coeffs), base::bits::SafeLog2Ceiling(num_coeffs)));
  }
```

**Review by Insun35 — APPROVED** [review]
LGTM

**Pull request merged**